nodejs/src/utils/QueryResultsIterator.ts

```diff
@@ -0,0 +1,23 @@
+import { QueryResults } from "../apis/interfaces/common";
+
+export class QueryResultsIterator<T> {
```

Comment:
this looks a pretty important class for a developer using the SDK. We should add very detailed tsdoc on how to use an object of this class(assuming that they will navigate here from their IDE)